linkerd/core/src/main/scala/io/buoyant/linkerd/AnnouncerInitializer.scala

```diff
@@ -0,0 +1,12 @@
+package io.buoyant.linkerd
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.twitter.finagle.Announcer
+import io.buoyant.config.ConfigInitializer
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "kind")
+trait AnnouncerConfig {
+  def mk(): Announcer
+}
+
+trait AnnouncerInitializer extends ConfigInitializer
```

Comment:
It would be helpful to have some documentation somewhere on how announcers work - it's sort of spread around the implementation right now and, unless I'm missing it, I don't really see an overview of how this works.  From what I gather, each server must be announced separately, and there is no mechanism for i.e. registering multiple servers in a single node?

Concrete example:
```yaml
- protocol: http
  servers:
  - label: https
    ...
  - label: http
    ...
  baseDtab: |
    /host/foo => /$/inet/127.1/8080 ;
    /host/bar => /$/inet/127.1/8081 ;
    /http/1.1/* => /host ;
```

I assume if I want to register both 'foo' and 'bar' i might have an announcers block like:
```yaml
- kind: io.l5d.zk.serversets
    hosts:
    - ...
    path: /role/env/foo
- kind: io.l5d.zk.serversets
    hosts:
    - ...
    path: /role/env/bar
```

But i'd end up with paths /role/env/bar/http, /role/env/bar/https, /role/env/foo/http, , /role/env/foo/https?  This might be okay, but it diverges a bit from, for instance, the announcer scheme used by aurora.